### ISSUE TITLE
Variable explorer index visual bug when sorting pandas dataframe

### ISSUE THREAD

**Issue by VShkaberda** [spyder-ide/spyder#14296]
## Problem Description

Index visually doesn't change when sorting pandas dataframe inside variable explorer. Doesn't matter if it's default index or not.

### What steps reproduce the problem?

```
import pandas as pd
df = pd.DataFrame({'month': [1, 4, 7, 10],
                   'year': [2012, 2014, 2013, 2014],
                   'sale': [55, 40, 84, 31]})
df.set_index('month', inplace=True)  # optional, default index has visual bug also
```

### What is the expected output? What do you see instead?
Expected output when sorting in older versions (example from Spyder 4.0.1):
![Expected](https://user-images.githubusercontent.com/11950122/99972096-c81b8580-2da6-11eb-8f35-b6faf81d29d1.PNG)
Output now:
![Now](https://user-images.githubusercontent.com/11950122/99972439-2fd1d080-2da7-11eb-8d41-a6810f9b32e8.PNG)
The dataframe is still sortable by index though index itself doesn't change visually:
![Sorted_by_index](https://user-images.githubusercontent.com/11950122/99972754-97881b80-2da7-11eb-89f3-b5e4a0865a93.PNG)

## Versions

* Spyder version: 4.1.5 (bug also exists in 4.2.0)
* Python version:  3.8.5
* Qt version: 5.9.7
* PyQt version: 5.9.2
* Operating System name/version: Windows 10

### Dependencies
```
# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.6.0 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.19.0 (OK)
jedi =0.17.1                   :  0.17.1 (OK)
nbconvert >=4.0                :  6.0.7 (OK)
numpydoc >=0.6.0               :  1.1.0 (OK)
paramiko >=2.4.0               :  2.7.2 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.2 (OK)
pygments >=2.0                 :  2.7.2 (OK)
pylint >=1.0                   :  2.6.0 (OK)
pyls >=0.34.0;<1.0.0           :  0.35.1 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  1.0.1 (OK)
qtconsole >=4.6.0              :  4.7.7 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.2.1 (OK)
spyder_kernels >=1.9.4;<1.10.0 :  1.9.4 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  19.0.2 (OK)

# Optional:
cython >=0.21                  :  None (OK)
matplotlib >=2.0.0             :  None (OK)
numpy >=1.7                    :  1.18.5 (OK)
pandas >=0.13.1                :  1.0.5 (OK)
scipy >=0.17.0                 :  1.3.2 (OK)
sympy >=0.7.3                  :  None (OK)
```

**Comment by rhkarls** [comment]
I think this is a duplicate of #13544 which was closed with #14259 , looks like its merged for the next release (4.2.1)

**Comment by ccordoba12** [comment]
Thanks @rhkarls for chiming in. What you said is correct: this problem is already fixed and the fix will be available in our 4.2.1 version, to be released in a month.

**Comment by VShkaberda** [comment]
My bad, I looked closely through opened issues. Thanks for the quick answer.
